pre_p2p/module.go

```diff
@@ -0,0 +1,53 @@
+package pre_p2p
```

Comment:
Justification for why this should be a TLD and not within P2P?

Reply:
Added this comment to the top of the file:

```
// TODO(team): This is a Top Level Module since it is a temporary parallel to the
// real `p2p` module. It should be removed once the real `p2p` module is ready but
// is meant to be a "real" replacement for now.
```

👍 ?

Reply:
But I feel as if it might be useful to keep a minimal mock of P2P indefinitely... The reason being, testing and simplicity. I believe all things P2P should live under the P2P dir.

Reply:
tl;dr Moved `pre2p` into `p2p`.

- I don't really see `pre2p` as a mock as much as it is an alternative implementation. 
- We'll also have REAL mocks with `mockgen` (or w/e we decide to use for mocks) 
- Ideally, the real P2P module should be able to replace pre2p 100%
- I don't feel too strongly whether it's a TLD or not, so moving into p2p